GVFS/FastFetch/NativeWindowsMethods.cs

```diff
@@ -0,0 +1,121 @@
+﻿using GVFS.Common.Tracing;
+using Microsoft.Win32.SafeHandles;
+using System;
+using System.ComponentModel;
+using System.IO;
+using System.IO.MemoryMappedFiles;
+using System.Runtime.InteropServices;
+
+namespace FastFetch
+{
+    public class NativeWindowsMethods
+    {
+        private const int AccessDeniedWin32Error = 5;
+
+        public static unsafe void WriteFile(ITracer tracer, byte* originalData, long originalSize, string destination, ushort mode /* ignored on windows */)
+        {
+            try
+            {
+                using (SafeFileHandle fileHandle = OpenForWrite(tracer, destination))
+                {
+                    if (fileHandle.IsInvalid)
+                    {
+                        throw new Win32Exception(Marshal.GetLastWin32Error());
+                    }
+
+                    byte* data = originalData;
+                    long size = originalSize;
+                    uint written = 0;
+                    while (size > 0)
+                    {
+                        uint toWrite = size < uint.MaxValue ? (uint)size : uint.MaxValue;
+                        if (!WriteFile(fileHandle, data, toWrite, out written, IntPtr.Zero))
+                        {
+                            throw new Win32Exception(Marshal.GetLastWin32Error());
+                        }
+
+                        size -= written;
+                        data = data + written;
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                tracer.RelatedError("Exception writing {0}: {1}", destination, e);
```

Comment:
Same comment here about using metadata and calling `ToString` (I realize the code might have been like this before but we might as well clean it up).